TaskfMRIAnalysis/scripts/makeSubjectTaskSummary.sh

```diff
@@ -0,0 +1,363 @@
+else
+	LowPassSTRING=""
+fi
+
+# Set variables used for different registration procedures
+if [ "${RegName}" != "NONE" ] ; then
+	RegString="_${RegName}"
+else
+	RegString=""
+fi
+
+if [[ "${SummaryName}" = "NONE" || "${SummaryName}" = "" ]]; then
+	if [ "${LevelTwofMRIName}" = "NONE" ]; then
+	    log_Err_Abort "Cannot determine summaryname. You must provide name for single-subject summary directory if Level2 analysis is not being run."
+	else
+		SummaryName="${LevelTwofMRIName}/${LevelTwofsfName}";
+	fi
+fi
+
+log_Msg "MAIN: SET_NAME_STRINGS: SmoothingString: ${SmoothingString}"
+log_Msg "MAIN: SET_NAME_STRINGS: TemporalFilterString: ${TemporalFilterString}"
+log_Msg "MAIN: SET_NAME_STRINGS: RegString: ${RegString}"
+
+
+# Determine locations of necessary directories (using expected naming convention)
+ResultsFolder="${Path}/${Subject}/MNINonLinear/Results"
+
+# Determine which analyses need to be packaged into summary directory
+# initialize run variables
+runParcellated=false; runVolume=false; runDense=false;
+Analyses=""; ExtensionList=""; ScalarExtensionList="";
+
+# Determine whether to run Parcellated, and set strings used for filenaming
+if [ "${Parcellation}" != "NONE" ] ; then
+	# Run Parcellated Analyses
+	runParcellated=true;
+	ParcellationString="_${Parcellation}"
+	Extension="ptseries.nii"
+	ScalarExtensionList="${ScalarExtensionList}pscalar.nii "
+	Analyses="${Analyses}ParcellatedStats "; # space character at end to separate multiple analyses
+fi
+
+# Determine whether to run Dense, and set strings used for filenaming
+if [ "${Parcellation}" = "NONE" ]; then
+	# Run Dense Analyses
+	runDense=true;
+	ParcellationString=""
+	Extension="dtseries.nii"
+	ExtensionList="${ExtensionList}dtseries.nii "
+	ScalarExtensionList="${ScalarExtensionList}dscalar.nii "
+	Analyses="${Analyses}GrayordinatesStats "; # space character at end to separate multiple analyses
+fi
+
+# Determine whether to run Volume, and set strings used for filenaming
+if [ "$VolumeBasedProcessing" = "YES" ] ; then
+	runVolume=true;
+	ExtensionList="${ExtensionList}nii.gz "
+	ScalarExtensionList="${ScalarExtensionList}volume.dscalar.nii "
+	Analyses="${Analyses}StandardVolumeStats "; # space character at end to separate multiple analyses	
+fi
+
+
+# Determine location where SummaryDirectory should be created
+SummaryDirectory="${ResultsFolder}/${SummaryName}${TemporalFilterString}${SmoothingString}${RegString}${ProcSTRING}${LowPassSTRING}${ParcellationString}_subjectSummary.feat"
+
+# Check if summary directory already exists, and remove old summary directory
+if [ -d "$SummaryDirectory" ]; then
+  echo "Removing previous summary directory from ${SummaryDirectory}..."
+  rm -r "$SummaryDirectory"
+fi
+
+# Create new summary directory
+echo "Creating new summary directory at ${SummaryDirectory}..."
+mkdir -v $SummaryDirectory
+
+# Create log file with arguments from TaskfMRIAnalysis.sh call
+touch ${SummaryDirectory}/TaskfMRIAnalysis.log
```

Comment:
Touch doesn't delete if the log happens to exist, and `>>` and similar don't care whether the file exists, this line is basically useless.  I don't think we want scripts to be guessing the name of log files other scripts are using, if that is what is going on here, as that would be inconvenient to edit properly.  I would say either pass in the name of the file to concatenate onto (if that is what we want), or name the hardcoded output according to the current (sub)script's name (and keep it separate from other scripts' outputs).

Reply:
@coalsont I'm not sure what you're suggesting. 
- This file is a log written by this script only. (Was there a reason you thought it was coming from another script?)
- I can get rid of the `touch` command. (I thought I had come across cases where an initial 'touch' was needed before using `>>` . Since I can recreate that situation here, I'm fine removing it.)
- But, I don't know what you mean by "name the hardcoded output according to the current (sub)script's name".  I want the name to be the same as the primary pipeline script, not this subscript. Even though the log is written by this subscript, it is summarizing what the main TaskfMRIAnalysis pipeline did altogether.

Reply:
The fact that you only used `>>` means it will add to whatever happens to already be in the file.  If you don't want any previous content, do `rm -f` to delete it instead of `touch`.  You should not need to `touch` before `>>` in any circumstance I can think of.

A log named `TaskfMRIAnalysis.log` makes me expect it was written by a `TaskfMRIAnalysis.sh` script, which is part of why I wasn't sure what you were doing with `touch` and `>>`.  I haven't looked at the details of what it contains, or what other logs this script makes, but I was confused by the name, so I will argue for renaming it.  In fact, ".log" implies it was generated while running the script it names, so that contributes to the confusion.  Something like `TaskAnalysisSummary.txt` seems more appropriate for "after the fact summarizing of other processing".

Reply:
Removed the touch commands in next commit. (However, kept the file name the same.)

Reply:
Actually, `TaskAnalysisSummary.txt` sounds good, so I'll change that in the next commit.